### DISCUSSION TITLE
How to recognize the soft return and hard return?

### DISCUSSION THREAD

**Discussion by fitexmage** [jsvine/pdfplumber#971] [Ask for help with specific PDFs]
I would like to combine the texts separated by soft return but keep the hard return.
<img width="815" alt="image" src="https://github.com/jsvine/pdfplumber/assets/29292288/21641afc-7596-455c-8c86-ad24f136cc0d">
For example, in this PDF, I want to extract the text to 
`额外参数：\n-b : extra_post_body 额外增加的请求内容,如-b "isp:001#002",抓包会用到，平时不会用到，如果value值为数组值，则用 # 进行分隔元素，也可以使用 # 强制生成数组类型如：001#`
which will keep the line break between line 1 and 2 but ignore the line break between line 2 and 3.
Is there any way to handle this?

**Comment by jsvine** [comment]
Hi @fitexmage, and thank you for you interest in `pdfplumber`. Generally speaking, PDFs don't have a concept of line breaks; instead, text is laid out positionally. If one line appears below another, it is because the PDF instructions indicate that lower position, rather than the PDF instructions including a newline command.

However, it is possible that, for your specific PDF, there is enough information to distinguish between these different transitions between lines. The first thing I'd suggest is examining the results of `page.extract_text_lines(...)`, and seeing whether there is a difference in the vertical distances between those lines (by examining the `top` and/or `bottom` values).

**Reply by fitexmage** [reply]
Thank you for your response! I have tried `page.extract_text_lines(...)` and found that the distance between line 1 and 2 is 2.349999999999909. The distance between line 2 and 3 is 2.3700000000000045. I also tried other lines and found the distances are around 2.36 for both soft return and hard return. It seems not have any patterns.
